java/maven/src/org/netbeans/modules/maven/execute/AbstractMavenExecutor.java

```diff
@@ -141,15 +142,13 @@ protected final void processInitialMessage() {
     }
 
     protected final void actionStatesAtStart() {
-        SwingUtilities.invokeLater(new Runnable() {
-            @Override public void run() {
-                createNewTabActions();
-                tabContext.rerun.setEnabled(false);
-                tabContext.rerunDebug.setEnabled(false);
-                tabContext.overview.setRoot(null);
-                tabContext.resume.setFinder(null);
-                tabContext.stop.setEnabled(true);
-            }
+        updateUILater(false, () -> {
+            createNewTabActions();
```

Comment:
These are typical UI updating actions and it makes no sense to call them when the system is running in headless mode. Thanks for introducing the `updateUILater` (if ever) utility!